coins/monero/src/wallet/seed/classic/mod.rs

```diff
@@ -0,0 +1,250 @@
+
+    let pos = i * 4;
+    result[pos .. pos + 4].copy_from_slice(&(indices[0] as u32).to_le_bytes());
+  }
+
+  Ok(result)
+}
+
+pub fn bytes_to_words(key: &[u8; 32], lang_name: LanguageName) -> Result<String, SeedError> {
+  // Grab the language, we can call unwrap because all language names must exist.
+  let lang = languages::all()?.into_iter().find(|l| l.language_name == lang_name).unwrap();
+
+  // get the language words
+  let words = &lang.word_list;
+  let word_list_length = words.len() as u32;
+
+  // To store the found words & add the checksum word later.
+  let mut result = vec![];
+
+  // convert to words
+  // 4 bytes -> 3 words. 8 digits base 16 -> 3 digits base 1626
+  for i in 0 .. 8 {
+    let mut indices: [u32; 4] = [0; 4];
+    // convert first 4 byte to u32 & get the word indices
+    let start = i * 4;
+    // convert 4 byte to u32
+    indices[0] = u32::from_le_bytes(key[start .. start + 4].try_into().unwrap());
+    indices[1] = indices[0] % word_list_length;
+    indices[2] = (indices[0] / word_list_length + indices[1]) % word_list_length;
+    indices[3] =
+      (((indices[0] / word_list_length) / word_list_length) + indices[2]) % word_list_length;
+
+    // append words to seed
+    result.push(words[indices[1] as usize].clone());
+    result.push(words[indices[2] as usize].clone());
+    result.push(words[indices[3] as usize].clone());
+  }
+  let checksum_index = create_checksum_index(&result, &lang)?;
+  result.push(result[checksum_index].clone());
+
+  Ok(result.join(" "))
+}
+
+/// takes a seed and returns the code for matching language
+fn find_seed_language(
+  seed: &[String],
+  has_checksum: bool,
+) -> Result<(Vec<usize>, Language), SeedError> {
+  let lang_list = languages::all()?;
+  let mut matched_indices = vec![];
+  let matched_language: Language;
+
+  for lang in &lang_list {
+    let word_map = &lang.word_map;
+    let trimmed_map = &lang.trimmed_word_map;
+
+    // Iterate through all the words and see if they're all present
+    let mut full_match = true;
+    for w in seed {
+      if has_checksum {
+        let trimmed_word = utf8_prefix(w, lang.unique_prefix_length);
+        if !trimmed_map.contains_key(&trimmed_word) {
+          full_match = false;
+          break;
+        }
+        matched_indices.push(trimmed_map[&trimmed_word]);
+      } else {
+        if !word_map.contains_key(w) {
+          full_match = false;
+          break;
+        }
+        matched_indices.push(word_map[w]);
+      }
+    }
+
+    if full_match && has_checksum && !checksum_test(seed, lang)? {
+      full_match = false;
+    }
+
+    if full_match {
+      matched_language = lang.clone();
+      return Ok((matched_indices, matched_language));
+    }
+
+    // no match for this language clear the indices
+    matched_indices.clear();
+  }
+
+  Err(SeedError::UnknownLanguage)?
+}
+
+fn utf8_prefix(word: &str, mut prefix_len: usize) -> String {
+  let mut prefix = String::new();
+  for a in word.chars() {
```

Comment:
This isn't returning bytes yet Unicode scalars. Those may be what you want already? If they're not, and you do want bytes, use `as_bytes`.